### PR TITLE
Virtual IP listener

### PR DESCRIPTION
**Pull request by cchristiezen**

### CODE REVIEW

File: zzk/service/hostregistry.go [hunk 1]
```diff
@@ -259,6 +267,8 @@ func (h *HostRegistryListener) Spawn(cancel <-chan interface{}, hostid string) {
 
 						// We have exceeded the wait timeout, so reschedule as
 						// soon as possible.
+						h.handler.UnassignAll(h.poolid, hostid)
```
Comment (smousa) [inline zzk/service/hostregistry.go]: move this to line 276

File: zzk/service/mocks/Connection.go [hunk 2]
```diff
@@ -0,0 +1,352 @@
+// Copyright 2017 The Serviced Authors.
```
Comment (smousa) [inline zzk/service/mocks/Connection.go]: Are you using this mock?

Comment (cchristiezen) [inline zzk/service/mocks/Connection.go]: I am.  The pool listener tests use it.

File: zzk/service/hostregistry.go (hunk 1 above)
Comment (cchristiezen) [inline zzk/service/hostregistry.go]: Moved.

**Pull request merged**